core/block.js

```diff
@@ -316,16 +316,14 @@ Blockly.Block.prototype.dispose = function(healStack) {
     this.inputList.length = 0;
     // Dispose of any remaining connections (next/previous/output).
     var connections = this.getConnections_(true);
-    for (var i = 0; i < connections.length; i++) {
-      var connection = connections[i];
-      if (connection.isConnected()) {
-        connection.disconnect();
-      }
-      connections[i].dispose();
+    for (var i = 0, connection; connection = connections[i]; i++) {
+      connection.dispose();
     }
   } finally {
     Blockly.Events.enable();
   }
+
+  this.disposed = true;
```

Comment:
Should this be inside the finally? I'm not sure what would happen with a half-disposed block, but I suspect it'd be better to mark it disposed.

Reply:
I think you're right, but I'm not sure. I'm planning to investigate this more once I get to block disposal.

The thing is I'm not sure exactly what that try block is catching; I'm thinking it might be for if removeInput throws? In which case it would be redundant (opt_quite could just be set to true instead).

For now I'm going to leave it (since it's just for testing), but this is the first thing I'm planning to scrub once I get to blocks!

Reply:
Fixed!